astropy/table/operations.py

```diff
@@ -87,9 +88,238 @@ def _get_out_class(objs):
     return out_class
 
 
+def join_skycoord(distance, distance_kind='sky'):
```

Comment:
I see your point, but passing in the function is harder for the user (extra import required).  We have had this philosophical difference going back to forever, but in the end we agreed that passing in strings should generally be allowed instead of requiring imported objects.

How about meeting in the middle and allowing a string which must correspond to a function in `coordinates`i.e. (try `getattr(astropy.coordinates, distance_func)`, or a function that has the same API as the current search_around functions.  So instead `'sky'` or `'3d'` it would be the full name.

Reply:
And I'll say that I am usually against putting in functionality where we have not yet identified a realistic use case.  In this case allowing passing in a user-defined distance function that meets the search_around API exactly.  So a plan B is to not allow passing in a function now and wait for user to request that feature.  And in fact in that case the user could make their own `join_func`.

Reply:
Yea, philosophy. For completeness: This doesn't require another import if `import astropy.coordinates as coord` is already done (pass `coord.search_around_sky`). And the (weak) argument against string values is that it requires reading to see what values are allowed - if the default was `coord.search_around_sky`, the user might think to try `coord.search_<TAB>` (in a notebook) to see what other options there are. But, yes, this is a weak argument and perhaps too "behavioral"...

In practice, though, I don't feel very strongly about this, so I leave it up to you!

Reply:
Yes, in my world I just do `from astropy.coordinates import SkyCoord` and that is all I need!  :smile:  And in practice time is running out ... I wasn't totally happy with `distance_kind` and I like `distance_func` better.

Reply:
Also, if the default of the function is based on the unit, this will very rarely be needed!